// @prometheus-io/lezer-promql/dist/index.es.js
import { LRParser } from '@lezer/lr';
import { styleTags, tags } from '@lezer/highlight';

// This file was generated by lezer-generator. You probably shouldn't edit it.
const inf$1 = 153,
  nan$1 = 154,
  Bool$1 = 1,
  Ignoring$1 = 2,
  On$1 = 3,
  GroupLeft$1 = 4,
  GroupRight$1 = 5,
  Offset$1 = 6,
  Atan2$1 = 7,
  Avg$1 = 8,
  Bottomk$1 = 9,
  Count$1 = 10,
  CountValues$1 = 11,
  Group$1 = 12,
  Max$1 = 13,
  Min$1 = 14,
  Quantile$1 = 15,
  Stddev$1 = 16,
  Stdvar$1 = 17,
  Sum$1 = 18,
  Topk$1 = 19,
  By$1 = 20,
  Without$1 = 21,
  And$1 = 22,
  Or$1 = 23,
  Unless$1 = 24,
  Start$1 = 25,
  End$1 = 26;

// Copyright 2021 The Prometheus Authors

const keywordTokens = {
    inf: inf$1,
    nan: nan$1,
    bool: Bool$1,
    ignoring: Ignoring$1,
    on: On$1,
    group_left: GroupLeft$1,
    group_right: GroupRight$1,
    offset: Offset$1,
};

const specializeIdentifier = (value, stack) => {
    return keywordTokens[value.toLowerCase()] || -1;
};

const contextualKeywordTokens = {
    avg: Avg$1,
    atan2: Atan2$1,
    bottomk: Bottomk$1,
    count: Count$1,
    count_values: CountValues$1,
    group: Group$1,
    max: Max$1,
    min: Min$1,
    quantile: Quantile$1,
    stddev: Stddev$1,
    stdvar: Stdvar$1,
    sum: Sum$1,
    topk: Topk$1,
    by: By$1,
    without: Without$1,
    and: And$1,
    or: Or$1,
    unless: Unless$1,
    start: Start$1,
    end: End$1,
};

const extendIdentifier = (value, stack) => {
    return contextualKeywordTokens[value.toLowerCase()] || -1;
};

// Copyright 2022 The Prometheus Authors

const promQLHighLight = styleTags({
    LineComment: tags.comment,
    LabelName: tags.labelName,
    StringLiteral: tags.string,
    NumberLiteral: tags.number,
    Duration: tags.number,
    Identifier: tags.variableName,
    'Abs Absent AbsentOverTime Acos Acosh Asin Asinh Atan Atanh AvgOverTime Ceil Changes Clamp ClampMax ClampMin Cos Cosh CountOverTime DaysInMonth DayOfMonth DayOfWeek DayOfYear Deg Delta Deriv Exp Floor HistogramAvg HistogramCount HistogramFraction HistogramQuantile HistogramSum HoltWinters Hour Idelta Increase Irate LabelReplace LabelJoin LastOverTime Ln Log10 Log2 MaxOverTime MinOverTime Minute Month Pi PredictLinear PresentOverTime QuantileOverTime Rad Rate Resets Round Scalar Sgn Sin Sinh Sort SortDesc SortByLabel SortByLabelDesc Sqrt StddevOverTime StdvarOverTime SumOverTime Tan Tanh Time Timestamp Vector Year':
        tags.function(tags.variableName),
    'Avg Bottomk Count Count_values Group Max Min Quantile Stddev Stdvar Sum Topk': tags.operatorKeyword,
    'By Without Bool On Ignoring GroupLeft GroupRight Offset Start End': tags.modifier,
    'And Unless Or': tags.logicOperator,
    'Sub Add Mul Mod Div Atan2 Eql Neq Lte Lss Gte Gtr EqlRegex EqlSingle NeqRegex Pow At': tags.operator,
    UnaryOp: tags.arithmeticOperator,
    '( )': tags.paren,
    '[ ]': tags.squareBracket,
    '{ }': tags.brace,
    '⚠': tags.invalid,
});

// This file was generated by lezer-generator. You probably shouldn't edit it.
const spec_Identifier = {__proto__:null,absent_over_time:325, absent:327, abs:329, acos:331, acosh:333, asin:335, asinh:337, atan:339, atanh:341, avg_over_time:343, ceil:345, changes:347, clamp:349, clamp_max:351, clamp_min:353, cos:355, cosh:357, count_over_time:359, days_in_month:361, day_of_month:363, day_of_week:365, day_of_year:367, deg:369, delta:371, deriv:373, exp:375, floor:377, histogram_count:379, histogram_fraction:381, histogram_quantile:383, histogram_stddev:385, histogram_stdvar:387, histogram_sum:389, histogram_avg:391, holt_winters:393, hour:395, idelta:397, increase:399, irate:401, label_replace:403, label_join:405, last_over_time:407, ln:409, log10:411, log2:413, mad_over_time:415, max_over_time:417, min_over_time:419, minute:421, month:423, pi:425, predict_linear:427, present_over_time:429, quantile_over_time:431, rad:433, rate:435, resets:437, round:439, scalar:441, sgn:443, sin:445, sinh:447, sort:449, sort_desc:451, sort_by_label:453, sort_by_label_desc:455, sqrt:457, stddev_over_time:459, stdvar_over_time:461, sum_over_time:463, tan:465, tanh:467, timestamp:469, time:471, vector:473, year:475};
const parser = LRParser.deserialize({
  version: 14,
  states: "8fOYQPOOO'kQPOOOOQO'#Cz'#CzO'pQPO'#CyQ'{QQOOOOQO'#Da'#DaO'vQPO'#D`OOQO'#FT'#FTO)YQPO'#FZOYQPO'#FVOYQPO'#FYO/tQSO'#F]O/|QQO'#F[OOQO'#F['#F[OOQO'#Fo'#FoOOQO'#Db'#DbOOQO'#Dd'#DdOOQO'#De'#DeOOQO'#Df'#DfOOQO'#Dg'#DgOOQO'#Dh'#DhOOQO'#Di'#DiOOQO'#Dj'#DjOOQO'#Dk'#DkOOQO'#Dl'#DlOOQO'#Dm'#DmOOQO'#Dn'#DnOOQO'#Do'#DoOOQO'#Dp'#DpOOQO'#Dq'#DqOOQO'#Dr'#DrOOQO'#Ds'#DsOOQO'#Dt'#DtOOQO'#Du'#DuOOQO'#Dv'#DvOOQO'#Dw'#DwOOQO'#Dx'#DxOOQO'#Dy'#DyOOQO'#Dz'#DzOOQO'#D{'#D{OOQO'#D|'#D|OOQO'#D}'#D}OOQO'#EO'#EOOOQO'#EP'#EPOOQO'#EQ'#EQOOQO'#ER'#EROOQO'#ES'#ESOOQO'#ET'#ETOOQO'#EU'#EUOOQO'#EV'#EVOOQO'#EW'#EWOOQO'#EX'#EXOOQO'#EY'#EYOOQO'#EZ'#EZOOQO'#E['#E[OOQO'#E]'#E]OOQO'#E^'#E^OOQO'#E_'#E_OOQO'#E`'#E`OOQO'#Ea'#EaOOQO'#Eb'#EbOOQO'#Ec'#EcOOQO'#Ed'#EdOOQO'#Ee'#EeOOQO'#Ef'#EfOOQO'#Eg'#EgOOQO'#Eh'#EhOOQO'#Ei'#EiOOQO'#Ej'#EjOOQO'#Ek'#EkOOQO'#El'#ElOOQO'#Em'#EmOOQO'#En'#EnOOQO'#Eo'#EoOOQO'#Ep'#EpOOQO'#Eq'#EqOOQO'#Er'#ErOOQO'#Es'#EsOOQO'#Et'#EtOOQO'#Eu'#EuOOQO'#Ev'#EvOOQO'#Ew'#EwOOQO'#Ex'#ExOOQO'#Ey'#EyOOQO'#Ez'#EzOOQO'#E{'#E{OOQO'#E|'#E|OOQO'#E}'#E}OOQO'#FO'#FOOOQO'#FP'#FPOOQO'#FQ'#FQQOQPOOO1gQPO'#C{O1lQPO'#DOO'vQPO,59eO1sQQO,59eO3aQPO,59kO3aQPO,59kO3aQPO,59kO3aQPO,59kO3aQPO,59kO9{QQO,5;mO:QQQO,5;pO:YQPO,5;}OOQO,59z,59zOOQO,5;o,5;oO:qQQO,5;qO:xQQO,5;tO<`QPO'#F^OOQO,5;w,5;wO<nQPO,5;wOOQO,5;v,5;vO<vQSO'#C|OOQO,59g,59gOOQO,59j,59jO=OQQO,59jOOQO1G/P1G/POOQO'#DR'#DRO1gQPO'#DSOOQO'#Fs'#FsO=YQPO'#FsOYQPO1G/VOYQPO1G/VOYQPO1G/VOYQPO1G/VOYQPO1G/VOCqQSO1G1XOOQO1G1[1G1[OCyQQO1G1[ODOQPO'#FTOOQO'#Fe'#FeOOQO1G1i1G1iODZQPO1G1iOOQO1G1]1G1]OOQO'#F_'#F_OD`QPO,5;xODeQSO1G1cODmQPO1G1cOOQO1G1c1G1cOOQO,59h,59hODuQPO,59hOYQPO'#FhOD}QPO1G/UOOQO1G/U1G/UOEVQPO,59nOOQO,5<_,5<_OLxQQO7+$qOMYQQO7+$qONnQQO7+$qO! UQQO7+$qO!!mQQO7+$qOOQO7+&s7+&sO!#WQQO7+&yOOQO7+&v7+&vO!#`QPO7+'TOOQO1G1d1G1dOOQO,5<T,5<TOOQO7+&}7+&}O!#eQSO7+&}OOQO-E9g-E9gO!#mQSO1G/SO!#uQPO1G/SOOQO1G/S1G/SO!#}QQO,5<SOOQO-E9f-E9fOOQO7+$p7+$pO!$XQPO1G/YOOQO<<Je<<JeO!*jQPO<<JeOOQO<<Jo<<JoOOQO<<Ji<<JiP!*oQSO'#FiOOQO,5<R,5<ROOQO7+$n7+$nO!*tQSO7+$nOOQO-E9e-E9eOOQO7+$t7+$tOOQOAN@PAN@POOQO<<HY<<HYP!*|QSO'#Fg",
  stateData: "!+R~O$bOSkOS~OWQOXQOYQOZQO[QO]QO^QO_QO`QOaQObQOcQO!V[O#z^O$_VO$`VO$dXO$h_O$i`O$jaO$kbO$lcO$mdO$neO$ofO$pgO$qhO$riO$sjO$tkO$ulO$vmO$wnO$xoO$ypO$zqO${rO$|sO$}tO%OuO%PvO%QwO%RxO%SyO%TzO%U{O%V|O%W}O%X!OO%Y!PO%Z!QO%[!RO%]!SO%^!TO%_!UO%`!VO%a!WO%b!XO%c!YO%d!ZO%e![O%f!]O%g!^O%h!_O%i!`O%j!aO%k!bO%l!cO%m!dO%n!eO%o!fO%p!gO%q!hO%r!iO%s!jO%t!kO%u!lO%v!mO%w!nO%x!oO%y!pO%z!qO%{!rO%|!sO%}!tO&O!uO&P!vO&Q!wO&R!xO&S!yO&T!zO&U!{O&V!|O&YWO&ZWO&[VO&^ZO~O!V!}O~Od#OOe#OO$d#PO~OU#YOV#SOf#VOg#WOh#VOt#SOw#SOx#SOy#SOz#TO{#TO|#UO}#UO!O#UO!P#UO!Q#UO!R#UO$W#ZO&W#XO~O$_#]O$`#]O&[#]OW#}XX#}XY#}XZ#}X[#}X]#}X^#}X_#}X`#}Xa#}Xb#}Xc#}X!V#}X#z#}X$_#}X$`#}X$d#}X$h#}X$i#}X$j#}X$k#}X$l#}X$m#}X$n#}X$o#}X$p#}X$q#}X$r#}X$s#}X$t#}X$u#}X$v#}X$w#}X$x#}X$y#}X$z#}X${#}X$|#}X$}#}X%O#}X%P#}X%Q#}X%R#}X%S#}X%T#}X%U#}X%V#}X%W#}X%X#}X%Y#}X%Z#}X%[#}X%]#}X%^#}X%_#}X%`#}X%a#}X%b#}X%c#}X%d#}X%e#}X%f#}X%g#}X%h#}X%i#}X%j#}X%k#}X%l#}X%m#}X%n#}X%o#}X%p#}X%q#}X%r#}X%s#}X%t#}X%u#}X%v#}X%w#}X%x#}X%y#}X%z#}X%{#}X%|#}X%}#}X&O#}X&P#}X&Q#}X&R#}X&S#}X&T#}X&U#}X&V#}X&Y#}X&Z#}X&[#}X&^#}X~Oq#`O&_#aO~O&^ZOU$OXV$OXf$OXg$OXh$OXt$OXw$OXx$OXy$OXz$OX{$OX|$OX}$OX!O$OX!P$OX!Q$OX!R$OX$W$OX$^$OX&W$OX$f$OX$e$OX~O$d#dO~O$f#fO~PYOd#OOe#OOUmaVmafmagmahmatmawmaxmaymazma{ma|ma}ma!Oma!Pma!Qma!Rma$Wma$^ma&Wma$fma$ema~OP#iOQ#jOR#jOW$gPX$gPY$gPZ$gP[$gP]$gP^$gP_$gP`$gPa$gPb$gPc$gP!V$gP#z$gP$_$gP$`$gP$d$gP$h$gP$i$gP$j$gP$k$gP$l$gP$m$gP$n$gP$o$gP$p$gP$q$gP$r$gP$s$gP$t$gP$u$gP$v$gP$w$gP$x$gP$y$gP$z$gP${$gP$|$gP$}$gP%O$gP%P$gP%Q$gP%R$gP%S$gP%T$gP%U$gP%V$gP%W$gP%X$gP%Y$gP%Z$gP%[$gP%]$gP%^$gP%_$gP%`$gP%a$gP%b$gP%c$gP%d$gP%e$gP%f$gP%g$gP%h$gP%i$gP%j$gP%k$gP%l$gP%m$gP%n$gP%o$gP%p$gP%q$gP%r$gP%s$gP%t$gP%u$gP%v$gP%w$gP%x$gP%y$gP%z$gP%{$gP%|$gP%}$gP&O$gP&P$gP&Q$gP&R$gP&S$gP&T$gP&U$gP&V$gP&Y$gP&Z$gP&[$gP&^$gP~O#v#rO~O{#tO#v#sO~Oi#vOj#vO$_VO$`VO&Y#uO&Z#uO&[VO~O$f#yO~P'{Ot#SOU#|aV#|af#|ag#|ah#|aw#|ax#|ay#|az#|a{#|a|#|a}#|a!O#|a!P#|a!Q#|a!R#|a$W#|a$^#|a&W#|a$f#|a$e#|a~O!R#zO$S#zO$T#zO$U#zO~O$e#|O&_$OO~Oq$QO$f$PO~O$e$RO$f$TO~P'{OQ#jOR#jOW$gXX$gXY$gXZ$gX[$gX]$gX^$gX_$gX`$gXa$gXb$gXc$gX!V$gX#z$gX$_$gX$`$gX$d$gX$h$gX$i$gX$j$gX$k$gX$l$gX$m$gX$n$gX$o$gX$p$gX$q$gX$r$gX$s$gX$t$gX$u$gX$v$gX$w$gX$x$gX$y$gX$z$gX${$gX$|$gX$}$gX%O$gX%P$gX%Q$gX%R$gX%S$gX%T$gX%U$gX%V$gX%W$gX%X$gX%Y$gX%Z$gX%[$gX%]$gX%^$gX%_$gX%`$gX%a$gX%b$gX%c$gX%d$gX%e$gX%f$gX%g$gX%h$gX%i$gX%j$gX%k$gX%l$gX%m$gX%n$gX%o$gX%p$gX%q$gX%r$gX%s$gX%t$gX%u$gX%v$gX%w$gX%x$gX%y$gX%z$gX%{$gX%|$gX%}$gX&O$gX&P$gX&Q$gX&R$gX&S$gX&T$gX&U$gX&V$gX&Y$gX&Z$gX&[$gX&^$gX~O&X$]O&]$^O~O#v$_O~O$_#]O$`#]O&[#]O~O$d$`O~O#z$aO~Oq#`O&_$cO~O$e$dO&_$cO~O$e$fO$f$hO~O$e$RO$f$kO~OS$lOT$lOWvaXvaYvaZva[va]va^va_va`vaavabvacva!Vva#zva$_va$`va$dva$hva$iva$jva$kva$lva$mva$nva$ova$pva$qva$rva$sva$tva$uva$vva$wva$xva$yva$zva${va$|va$}va%Ova%Pva%Qva%Rva%Sva%Tva%Uva%Vva%Wva%Xva%Yva%Zva%[va%]va%^va%_va%`va%ava%bva%cva%dva%eva%fva%gva%hva%iva%jva%kva%lva%mva%nva%ova%pva%qva%rva%sva%tva%uva%vva%wva%xva%yva%zva%{va%|va%}va&Ova&Pva&Qva&Rva&Sva&Tva&Uva&Vva&Yva&Zva&[va&^va~Ot#SOUsqfsqgsqhsqzsq{sq|sq}sq!Osq!Psq!Qsq!Rsq$Wsq$^sq&Wsq$fsq$esq~OVsqwsqxsqysq~PKnOV#SOw#SOx#SOy#SO~PKnOV#SOt#SOw#SOx#SOy#SOz#TO{#TOUsqfsqgsqhsq$Wsq$^sq&Wsq$fsq$esq~O|sq}sq!Osq!Psq!Qsq!Rsq~PMjO|#UO}#UO!O#UO!P#UO!Q#UO!R#UO~PMjOV#SOf#VOh#VOt#SOw#SOx#SOy#SOz#TO{#TO|#UO}#UO!O#UO!P#UO!Q#UO!R#UO~OUsqgsq$Wsq$^sq&Wsq$fsq$esq~P! lO#v$nO&X$mO~O$f$oO~Oq#`O&_$pO~Oq$rO$f$sO~O$e$tO$f$sO~O$e$[a$f$[a~P'{O$d#dOWviXviYviZvi[vi]vi^vi_vi`viavibvicvi!Vvi#zvi$_vi$`vi$hvi$ivi$jvi$kvi$lvi$mvi$nvi$ovi$pvi$qvi$rvi$svi$tvi$uvi$vvi$wvi$xvi$yvi$zvi${vi$|vi$}vi%Ovi%Pvi%Qvi%Rvi%Svi%Tvi%Uvi%Vvi%Wvi%Xvi%Yvi%Zvi%[vi%]vi%^vi%_vi%`vi%avi%bvi%cvi%dvi%evi%fvi%gvi%hvi%ivi%jvi%kvi%lvi%mvi%nvi%ovi%pvi%qvi%rvi%svi%tvi%uvi%vvi%wvi%xvi%yvi%zvi%{vi%|vi%}vi&Ovi&Pvi&Qvi&Rvi&Svi&Tvi&Uvi&Vvi&Yvi&Zvi&[vi&^vi~O&X$wO~Oq#`O~Oq$rO$f$xO~Oq$rO~O",
  goto: ")R$hPPPPPPPPPPPPPPPPPPPPPPPPPPPPP$i$u%R%XP%b$iP%k%rPPPPPPPPPPP$i%|&YP&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y&Y$iP&f$i$iP$i$i&u$i'R'b'jPPP$iP'mP'p'v'|PPPPP(SPPP(re^OXY#P#m#n#o#p#q$ReROXY#P#m#n#o#p#q$RQ#QRR#h#RQ#e#OQ$U#jR$v$lQ#RRQ#[UR#h#QZ#l#S#T#U#V#WY#k#S#T#U#V#WR$V#leUOXY#P#m#n#o#p#q$ReTOXY#P#m#n#o#p#q$Rd^OXY#P#m#n#o#p#q$RR#w#ZeYOXY#P#m#n#o#p#q$Rd]OXY#P#m#n#o#p#q$RR#c[Q#bZV$b#|$d$qR#{#`R#x#ZQ$g$QR$u$gQ$S#gR$j$SQ#}#bR$e#}QSOQ#^XQ#_YQ#g#PQ$W#mQ$X#nQ$Y#oQ$Z#pQ$[#qR$i$RQ#m#SQ#n#TQ#o#UQ#p#VR#q#W",
  nodeNames: "⚠ Bool Ignoring On GroupLeft GroupRight Offset Atan2 Avg Bottomk Count CountValues Group Max Min Quantile Stddev Stdvar Sum Topk By Without And Or Unless Start End LineComment PromQL AggregateExpr AggregateOp AggregateModifier GroupingLabels LabelName FunctionCallBody BinaryExpr Pow BoolModifier MatchingModifierClause Mul Div Mod Add Sub Eql Gte Gtr Lte Lss Neq FunctionCall FunctionIdentifier AbsentOverTime Identifier Absent Abs Acos Acosh Asin Asinh Atan Atanh AvgOverTime Ceil Changes Clamp ClampMax ClampMin Cos Cosh CountOverTime DaysInMonth DayOfMonth DayOfWeek DayOfYear Deg Delta Deriv Exp Floor HistogramCount HistogramFraction HistogramQuantile HistogramStdDev HistogramStdVar HistogramSum HistogramAvg HoltWinters Hour Idelta Increase Irate LabelReplace LabelJoin LastOverTime Ln Log10 Log2 MadOverTime MaxOverTime MinOverTime Minute Month Pi PredictLinear PresentOverTime QuantileOverTime Rad Rate Resets Round Scalar Sgn Sin Sinh Sort SortDesc SortByLabel SortByLabelDesc Sqrt StddevOverTime StdvarOverTime SumOverTime Tan Tanh Timestamp Time Vector Year MatrixSelector Duration NumberLiteral OffsetExpr ParenExpr StringLiteral SubqueryExpr UnaryExpr UnaryOp VectorSelector LabelMatchers LabelMatcher MatchOp EqlSingle EqlRegex NeqRegex StepInvariantExpr At AtModifierPreprocessors MetricName",
  maxTerm: 245,
  nodeProps: [
    ["group", -12,29,35,50,129,131,132,133,134,135,136,138,145,"Expr"]
  ],
  propSources: [promQLHighLight],
  skippedNodes: [0,27],
  repeatNodeCount: 3,
  tokenData: "3s~RwX^#lpq#lqr$ars$tst&luv'Twx'Yxy({yz)Qz{)V{|)[|})c}!O)h!O!P)o!P!Q*o!Q!R*t!R![+l![!]0W!^!_1S!_!`1a!`!a1v!b!c2T!c!}2Y!}#O2p#P#Q2u#Q#R2z#R#S2Y#S#T3P#T#o2Y#o#p3i#q#r3n#y#z#l$f$g#l#BY#BZ#l$IS$I_#l$I|$JO#l$JT$JU#l$KV$KW#l&FU&FV#l~#qY$b~X^#lpq#l#y#z#l$f$g#l#BY#BZ#l$IS$I_#l$I|$JO#l$JT$JU#l$KV$KW#l&FU&FV#l~$dQ!_!`$j#r#s$o~$oO!R~~$tO$U~~$yW#z~OY$tZr$trs%cs#O$t#O#P%h#P;'S$t;'S;=`&f<%lO$t~%hO#z~~%kRO;'S$t;'S;=`%t;=`O$t~%yX#z~OY$tZr$trs%cs#O$t#O#P%h#P;'S$t;'S;=`&f;=`<%l$t<%lO$t~&iP;=`<%l$t~&qSk~OY&lZ;'S&l;'S;=`&}<%lO&l~'QP;=`<%l&l~'YOy~~'_W#z~OY'YZw'Ywx%cx#O'Y#O#P'w#P;'S'Y;'S;=`(u<%lO'Y~'zRO;'S'Y;'S;=`(T;=`O'Y~(YX#z~OY'YZw'Ywx%cx#O'Y#O#P'w#P;'S'Y;'S;=`(u;=`<%l'Y<%lO'Y~(xP;=`<%l'Y~)QO$d~~)VO$f~~)[Ow~R)cO&ZPzQ~)hO$e~R)oO&YP{QP)rP!Q![)uP)zR&[P!Q![)u!g!h*T#X#Y*TP*WR{|*a}!O*a!Q![*gP*dP!Q![*gP*lP&[P!Q![*g~*tOx~R*yZ&[P!O!P)u!Q![+l!g!h*T#W#X,a#X#Y*T#[#],x#a#b-^#g#h.Q#k#l.c#l#m/l#m#n.}R+qY&[P!O!P)u!Q![+l!g!h*T#W#X,a#X#Y*T#[#],x#a#b-^#g#h.Q#k#l.c#m#n.}Q,fP#vQ!Q![,iQ,lS!Q![,i#[#],x#a#b-^#g#h.QQ,}P#vQ!Q![-QQ-TR!Q![-Q#a#b-^#g#h.QQ-cQ#vQ!Q![-i#g#h-{Q-lR!Q![-i#a#b-u#g#h.QQ-xP#g#h-{Q.QO#vQQ.VP#vQ!Q![.YQ.]Q!Q![.Y#a#b-uQ.hP#vQ!Q![.kQ.nT!Q![.k#W#X,a#[#],x#a#b-^#g#h.QQ/SP#vQ!Q![/VQ/YU!Q![/V#W#X,a#[#],x#a#b-^#g#h.Q#k#l.cP/oR!Q![/x!c!i/x#T#Z/xP/}R&[P!Q![/x!c!i/x#T#Z/xV0_T&]S!VR!Q![0n![!]0n!c!}0n#R#S0n#T#o0nR0sT!VR!Q![0n![!]0n!c!}0n#R#S0n#T#o0n~1XP!Q~!_!`1[~1aO!P~~1fQ$SP!_!`1l#r#s1qQ1qO|Q~1vO$T~~1{P!O~!_!`2O~2TO}~~2YO$W~V2aT!VRqS!Q![2Y![!]0n!c!}2Y#R#S2Y#T#o2Y~2uO&W~~2zO&X~~3POt~~3STO#S3P#S#T%c#T;'S3P;'S;=`3c<%lO3P~3fP;=`<%l3P~3nO&^~~3sO&_~",
  tokenizers: [0, 1, 2],
  topRules: {"PromQL":[0,28],"MetricName":[1,148]},
  specialized: [{term: 53, get: (value, stack) => (specializeIdentifier(value) << 1), external: specializeIdentifier},{term: 53, get: (value, stack) => (extendIdentifier(value) << 1) | 1, external: extendIdentifier, extend: true},{term: 53, get: (value) => spec_Identifier[value] || -1}],
  tokenPrec: 0
});
// This file was generated by lezer-generator. You probably shouldn't edit it.
const inf = 153,
  nan = 154,
  Bool = 1,
  Ignoring = 2,
  On = 3,
  GroupLeft = 4,
  GroupRight = 5,
  Offset = 6,
  Atan2 = 7,
  Avg = 8,
  Bottomk = 9,
  Count = 10,
  CountValues = 11,
  Group = 12,
  Max = 13,
  Min = 14,
  Quantile = 15,
  Stddev = 16,
  Stdvar = 17,
  Sum = 18,
  Topk = 19,
  By = 20,
  Without = 21,
  And = 22,
  Or = 23,
  Unless = 24,
  Start = 25,
  End = 26,
  LineComment = 27,
  PromQL = 28,
  AggregateExpr = 29,
  AggregateOp = 30,
  AggregateModifier = 31,
  GroupingLabels = 32,
  LabelName = 33,
  FunctionCallBody = 34,
  BinaryExpr = 35,
  Pow = 36,
  BoolModifier = 37,
  MatchingModifierClause = 38,
  Mul = 39,
  Div = 40,
  Mod = 41,
  Add = 42,
  Sub = 43,
  Eql = 44,
  Gte = 45,
  Gtr = 46,
  Lte = 47,
  Lss = 48,
  Neq = 49,
  FunctionCall = 50,
  FunctionIdentifier = 51,
  AbsentOverTime = 52,
  Identifier = 53,
  Absent = 54,
  Abs = 55,
  Acos = 56,
  Acosh = 57,
  Asin = 58,
  Asinh = 59,
  Atan = 60,
  Atanh = 61,
  AvgOverTime = 62,
  Ceil = 63,
  Changes = 64,
  Clamp = 65,
  ClampMax = 66,
  ClampMin = 67,
  Cos = 68,
  Cosh = 69,
  CountOverTime = 70,
  DaysInMonth = 71,
  DayOfMonth = 72,
  DayOfWeek = 73,
  DayOfYear = 74,
  Deg = 75,
  Delta = 76,
  Deriv = 77,
  Exp = 78,
  Floor = 79,
  HistogramCount = 80,
  HistogramFraction = 81,
  HistogramQuantile = 82,
  HistogramStdDev = 83,
  HistogramStdVar = 84,
  HistogramSum = 85,
  HistogramAvg = 86,
  HoltWinters = 87,
  Hour = 88,
  Idelta = 89,
  Increase = 90,
  Irate = 91,
  LabelReplace = 92,
  LabelJoin = 93,
  LastOverTime = 94,
  Ln = 95,
  Log10 = 96,
  Log2 = 97,
  MadOverTime = 98,
  MaxOverTime = 99,
  MinOverTime = 100,
  Minute = 101,
  Month = 102,
  Pi = 103,
  PredictLinear = 104,
  PresentOverTime = 105,
  QuantileOverTime = 106,
  Rad = 107,
  Rate = 108,
  Resets = 109,
  Round = 110,
  Scalar = 111,
  Sgn = 112,
  Sin = 113,
  Sinh = 114,
  Sort = 115,
  SortDesc = 116,
  SortByLabel = 117,
  SortByLabelDesc = 118,
  Sqrt = 119,
  StddevOverTime = 120,
  StdvarOverTime = 121,
  SumOverTime = 122,
  Tan = 123,
  Tanh = 124,
  Timestamp = 125,
  Time = 126,
  Vector = 127,
  Year = 128,
  MatrixSelector = 129,
  Duration = 130,
  NumberLiteral = 131,
  OffsetExpr = 132,
  ParenExpr = 133,
  StringLiteral = 134,
  SubqueryExpr = 135,
  UnaryExpr = 136,
  UnaryOp = 137,
  VectorSelector = 138,
  LabelMatchers = 139,
  LabelMatcher = 140,
  MatchOp = 141,
  EqlSingle = 142,
  EqlRegex = 143,
  NeqRegex = 144,
  StepInvariantExpr = 145,
  At = 146,
  AtModifierPreprocessors = 147,
  MetricName = 148;

export { Abs, Absent, AbsentOverTime, Acos, Acosh, Add, AggregateExpr, AggregateModifier, AggregateOp, And, Asin, Asinh, At, AtModifierPreprocessors, Atan, Atan2, Atanh, Avg, AvgOverTime, BinaryExpr, Bool, BoolModifier, Bottomk, By, Ceil, Changes, Clamp, ClampMax, ClampMin, Cos, Cosh, Count, CountOverTime, CountValues, DayOfMonth, DayOfWeek, DayOfYear, DaysInMonth, Deg, Delta, Deriv, Div, Duration, End, Eql, EqlRegex, EqlSingle, Exp, Floor, FunctionCall, FunctionCallBody, FunctionIdentifier, Group, GroupLeft, GroupRight, GroupingLabels, Gte, Gtr, HistogramAvg, HistogramCount, HistogramFraction, HistogramQuantile, HistogramStdDev, HistogramStdVar, HistogramSum, HoltWinters, Hour, Idelta, Identifier, Ignoring, Increase, Irate, LabelJoin, LabelMatcher, LabelMatchers, LabelName, LabelReplace, LastOverTime, LineComment, Ln, Log10, Log2, Lss, Lte, MadOverTime, MatchOp, MatchingModifierClause, MatrixSelector, Max, MaxOverTime, MetricName, Min, MinOverTime, Minute, Mod, Month, Mul, Neq, NeqRegex, NumberLiteral, Offset, OffsetExpr, On, Or, ParenExpr, Pi, Pow, PredictLinear, PresentOverTime, PromQL, Quantile, QuantileOverTime, Rad, Rate, Resets, Round, Scalar, Sgn, Sin, Sinh, Sort, SortByLabel, SortByLabelDesc, SortDesc, Sqrt, Start, Stddev, StddevOverTime, Stdvar, StdvarOverTime, StepInvariantExpr, StringLiteral, Sub, SubqueryExpr, Sum, SumOverTime, Tan, Tanh, Time, Timestamp, Topk, UnaryExpr, UnaryOp, Unless, Vector, VectorSelector, Without, Year, inf, nan, parser };
